lxd/db/query/transaction.go

```diff
@@ -12,8 +12,12 @@ import (
 
 // Transaction executes the given function within a database transaction with a 10s context timeout.
 func Transaction(ctx context.Context, db *sql.DB, f func(context.Context, *sql.Tx) error) error {
-	ctx, cancel := context.WithTimeout(ctx, time.Second*10)
-	defer cancel()
+	_, ok := ctx.Deadline()
```

Comment:
@markylaing btw, what was the trigger you had for needing this?

Reply:
My only concern with this is whether in merging this we will inadvertently increase the timeout used for certain queries in LXD because they are currently being passed a context with a longer deadline.

Reply:
The pattern we have in LXD Cloud is to always have a default API timeout applied to the request context. When running the cloud with a debugger I was just using a `context.Background()`, but was still finding that transactions were failing due to timeouts. I figure with this change I can set a high timeout when I need to and it will apply to the whole API route.

I can get around this by making the change locally and using a replace directive in the go.mod. So it's not a big deal.

Reply:
Hrm, are the long transactions not a concern then?

Reply:
In terms of what is currently in the code base, presumably no transactions are currently slower than 10 seconds else we'd see lots of test failures. So a higher timeout isn't necessarily a problem, but we might see issues for large clusters in future.